internal/dataplane/kong_client.go

```diff
@@ -518,13 +523,14 @@ func (c *KongClient) tryRecoveringWithFallbackConfiguration(
 	if err != nil {
 		return fmt.Errorf("failed to extract broken objects from update error: %w", err)
 	}
-	fallbackCache, err := c.fallbackConfigGenerator.GenerateExcludingAffected(
+	fallbackCache, newHash, err := c.fallbackConfigGenerator.GenerateExcludingBrokenObjects(
 		cacheSnapshot,
 		brokenObjects,
 	)
 	if err != nil {
 		return fmt.Errorf("failed to generate fallback configuration: %w", err)
 	}
+	c.kongConfig.SnapshotHash = newHash
```

Comment:
I think we should only update this hash when taking a snapshot in `KongClient.Update` initial phase as we need to keep a hash of a cache stores' snapshot we "processed" in the last `KongClient.Update` call.